addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InformationDisclosureSuspiciousCommentsScanRule.java

```diff
@@ -132,18 +154,19 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 Element el;
                 int offset = 0;
                 while ((el = source.getNextElement(offset, HTMLElementName.SCRIPT)) != null) {
-                    for (Pattern pattern : patterns) {
-                        String elStr = el.toString();
-                        Matcher m = pattern.matcher(elStr);
-                        if (m.find()) {
-                            recordAlertSummary(
-                                    alertMap,
-                                    new AlertSummary(
-                                            pattern.toString(),
-                                            elStr,
-                                            Alert.CONFIDENCE_LOW,
-                                            m.group()));
-                            break; // Only need to record this script once
+                    for (String candidate : getJsComments(el.toString())) {
+                        for (Pattern pattern : patterns) {
+                            Matcher m = pattern.matcher(candidate);
+                            if (m.find()) {
+                                recordAlertSummary(
+                                        alertMap,
+                                        new AlertSummary(
+                                                pattern.toString(),
+                                                candidate,
+                                                Alert.CONFIDENCE_LOW,
+                                                m.group()));
+                                break; // Only need to record this script once
+                            }
                         }
```

Comment:
This should be deduplicated (which also addresses the break not breaking on the expected loop anymore and comments lying).